optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/spotroster/SpotRosterViewport.java

```diff
@@ -0,0 +1,132 @@
+import java.util.stream.Stream;
+
+import org.jboss.errai.common.client.api.elemental2.IsElement;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.grid.CssGridLines;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.grid.Ticks;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.Blob;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.Lane;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.LinearScale;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.Orientation;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.SubLane;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.Viewport;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.view.BlobView;
+import org.optaplanner.openshift.employeerostering.shared.shift.Shift;
+import org.optaplanner.openshift.employeerostering.shared.spot.Spot;
+import org.optaplanner.openshift.employeerostering.shared.timeslot.TimeSlot;
+
+import static java.util.Collections.singletonList;
+import static org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.Orientation.HORIZONTAL;
+
+public class SpotRosterViewport extends Viewport<LocalDateTime> {
+
+    private final Integer tenantId;
+    private final Supplier<ShiftBlobView> blobViewSupplier;
+    private final LinearScale<LocalDateTime> scale;
+    private final CssGridLines gridLines;
+    private final Ticks<LocalDateTime> ticks;
+    private final List<Lane<LocalDateTime>> lanes;
+
+    SpotRosterViewport(final Integer tenantId,
+                       final Supplier<ShiftBlobView> blobViewSupplier,
+                       final LinearScale<LocalDateTime> scale,
+                       final CssGridLines gridLines,
+                       final Ticks<LocalDateTime> ticks,
+                       final List<Lane<LocalDateTime>> lanes) {
+
+        this.tenantId = tenantId;
+        this.blobViewSupplier = blobViewSupplier;
+        this.scale = scale;
+        this.gridLines = gridLines;
+        this.ticks = ticks;
+        this.lanes = lanes;
+    }
+
+    @Override
+    public void drawGridLinesAt(final IsElement target) {
+        gridLines.drawAt(target, this);
+    }
+
+    @Override
+    public void drawTicksAt(final IsElement target) {
+        //FIXME: Make it18n
+        ticks.drawAt(target, this, date -> {
+            final int hours = date.getHour();
+            if (hours == 0) {
+                final String lowerDayOfTheWeek = date.getDayOfWeek().toString().toLowerCase();
+                final String dayOfTheWeek = lowerDayOfTheWeek.substring(0, 1).toUpperCase() + lowerDayOfTheWeek.substring(1);
+                return dayOfTheWeek.substring(0, 3) + " " + date.getDayOfMonth();
+            } else {
+                return (hours < 10 ? "0" : "") + hours + ":00";
```

Comment:
Saw this code somewhere else. Could be good to refactor it into a utility method.

Reply:
Saw comment above. I don't think it's worth it to spend any time refactoring this at the moment